sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -521,16 +521,31 @@ class Analyzer(
    */
   object ResolveSortReferences extends Rule[LogicalPlan] {
     def apply(plan: LogicalPlan): LogicalPlan = plan resolveOperators {
-      case s @ Sort(ordering, global, p @ Project(projectList, child))
-          if !s.resolved && p.resolved =>
-        val (newOrdering, missing) = resolveAndFindMissing(ordering, p, child)
+      case s @ Sort(_, _, p @ Project(_, _)) if !s.resolved && p.resolved =>
+        val (newOrdering, missing, newChild): (Seq[SortOrder], Seq[Attribute], LogicalPlan) =
+          p.child match {
+            // Case 1: when WINDOW functions are used in the SELECT clause.
+            //   Example: SELECT sum(col1) OVER() FROM table1 ORDER BY col2
+            case p1 @ Project(_, w @ Window(_, _, _, _, p2: Project)) =>
```

Comment:
What if the child of Window is not a Project (a Aggregation or other)? Or there are multiple projections ?